src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (isEmpty()) {
+            newNode = new Node<>(null, value, null);
+            tail = newNode;
+        } else {
+            newNode = new Node<>(head, value, null);
+            head.next = newNode;
+        }
+        head = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> current = getNode(index);
+        Node<T> newNode = new Node<>(current.prev, value, current);
+        if (current.prev != null) {
+            current.prev.next = newNode;
+        } else {
+            tail = newNode;
+        }
+        current.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = getNode(index);
+        T oldItem = current.element;
+        current.element = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> current = getNode(object);
```

Comment:
You can avoid creating getNode(object) method
```suggestion
        Node<T> current = head;
        while (current != null) {
            if (current.element == object || current.element != null && current.element.equals(object)) {
                unlink(current);
                return true;
            }
            current = current.next;
        }
        return false;
```